components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/logout/validators/LogoutReqSignatureValidator.java

```diff
@@ -0,0 +1,256 @@
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants.SIGNATURE;
+import static org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants.SIGNATURE_ALGORITHM;
+import static org.wso2.carbon.identity.application.authenticator.samlsso.util.SSOConstants.HTTP_POST_PARAM_SAML2_AUTH_REQ;
+import static org.wso2.carbon.identity.application.authenticator.samlsso.util.SSOConstants.RELAY_STATE;
+
+/**
+ * This class is used to validate the signature in SAML logout Request.
+ */
+public class LogoutReqSignatureValidator {
+
+    private static final Log log = LogFactory.getLog(LogoutReqSignatureValidator.class);
+
+    /**
+     * Validates the signature of the SAML requests sent with HTTP Redirect Binding against the given certificate.
+     *
+     * @param queryString SAML request (passed an an HTTP query parameter).
+     * @param issuer      Issuer of the SAML request.
+     * @param certificate Certificate for validating the signature
+     * @return true       If the signature is valid, false otherwise.
+     * @throws SecurityException If signature validation process fails.
+     */
+    public boolean validateSignature(String queryString, String issuer, X509Certificate certificate)
+            throws SecurityException, IdentityException {
+
+        byte[] signature = getSignature(queryString);
+        byte[] signedContent = getSignedContent(queryString);
+        String algorithmUri = getSignatureAlgorithm(queryString);
+        CriteriaSet criteriaSet = buildCriteriaSet(issuer);
+
+        // creating the SAML2HTTPRedirectDeflateSignatureRule
+        X509CredentialImpl credential = new X509CredentialImpl(certificate, issuer);
+
+        List<Credential> credentials = new ArrayList<Credential>();
+        credentials.add(credential);
+        CollectionCredentialResolver credentialResolver = new CollectionCredentialResolver(credentials);
+        KeyInfoCredentialResolver keyResolver = DefaultSecurityConfigurationBootstrap.
+                buildBasicInlineKeyInfoCredentialResolver();
+        SignatureTrustEngine engine = new ExplicitKeySignatureTrustEngine(credentialResolver, keyResolver);
+        return engine.validate(signature, signedContent, algorithmUri, criteriaSet, null);
+    }
+
+    /**
+     * Validate the  Signature in the SAML Assertion.
+     *
+     * @param request SAML Assertion (SAML LogoutRequest).
+     * @param cred    Signature signing credential.
+     * @param alias   Certificate alias against which the signature is validated.
+     * @return true   If the signature is valid.
+     * @throws IdentityException If signature validation process fails.
+     */
+    public boolean validateXMLSignature(SignableXMLObject request, X509Credential cred,
+                                        String alias) throws IdentityException {
```

Comment:
any specific reason for having the `alias` ?

Reply:
Fixed with PR #133